### DISCUSSION TITLE
Get more than 100 records with a.protocol.resolve_names

### DISCUSSION THREAD

**Discussion by hishamahmeddeep** [ecederstrand/exchangelib#1312] [General]
I am able to get 100 records with a.protocol.resolve_names
but how can i get next 100 or more records?

is there a way to get exact name search? with any other class methods?

**Comment by ecederstrand** [comment]
EWS does not support returning more than 100 results. See https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/resolvenames-operation

You can match against a number of fields on the contact, which should allow you to refine your search to be more specific: https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/unresolvedentry

**Comment by hishamahmeddeep** [comment]
thnx @ecederstrand 
i already knew that it doesnt provide more than 100 results
thnx again for pointing the refine search

is above possible with your library?

**Reply by hishamahmeddeep** [reply]
how can I refine search ?  @ecederstrand 
an example will be highly appreciated 
thnx

**Reply by ecederstrand** [reply]
That depends on what you want to search for and which hint you can give to ResolveNames. Only you known that.

I assume if you search for an SMTP address, for example, then at most a few results would be returned.

**Reply by hishamahmeddeep** [reply]
@ecederstrand 

can i use a.protocol.resolve_names to match against a number of fields on the contact ?
i need to use resolve_names because the data is coming from Active directory , not user's own contract
i need to search for say : FirstName / group name for starters. 

thnx for your support

**Reply by ecederstrand** [reply]
`Protocol.resolve_names()` is just an interface for the `ResolveNames` operation: https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/resolvenames-operation

I don't have more knowledge about that service other than what is described in that documentation.
